QModManager/Patching/GameDetector.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace QModManager.Patching
+{
+    using System;
+    using System.IO;
+    using QModManager.API;
+    using QModManager.Utility;
+
+    internal class GameDetector
+    {
+        internal readonly Game CurrentlyRunningGame;
+
+        internal bool IsValidGameRunning => CurrentlyRunningGame == Game.Subnautica || CurrentlyRunningGame == Game.BelowZero;
+
+        internal GameDetector()
+        {
+            bool isSubnautica = Directory.GetFiles(Environment.CurrentDirectory, "Subnautica.exe", SearchOption.TopDirectoryOnly).Length > 0
+                || Directory.GetFiles(Environment.CurrentDirectory, "Subnautica.app", SearchOption.TopDirectoryOnly).Length > 0
+                || Directory.GetDirectories(Environment.CurrentDirectory, "Subnautica.app", SearchOption.TopDirectoryOnly).Length > 0;
```

Comment:
`.app` is a directory extension, not a file extension. The `GetFiles` call can be safely removed